index.js

```diff
@@ -45,6 +45,41 @@ app.get('/', function(req, res) {
   res.render('index');
 });
 
+app.get('/devices/:token', function(req, res) {
+  return sendMachines(req, res, req.params.token);
+});
+
+// get machines for the token and send them along with the token
+function sendMachines(req, res, token) {
+  if (!token) {
+    throw('No token provided');
```

Comment:
`throw new Error('No token provided')` should improve the error message when this happens.